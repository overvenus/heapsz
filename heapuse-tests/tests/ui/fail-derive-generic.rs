use heapuse_derive::Heap;
use heapuse_example::*;

#[derive(Heap)]
pub struct G1<T> {
    #[heap(add)]
    pub a: T,
}

fn main() {}
