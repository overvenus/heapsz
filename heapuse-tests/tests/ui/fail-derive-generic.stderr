error[E0277]: the trait bound `T: HeapSize` is not satisfied
 --> tests/ui/fail-derive-generic.rs:5:5
  |
5 | /     #[heap_size]
6 | |     pub a: T,
  | |____________^ the trait `HeapSize` is not implemented for `T`
  |
help: consider restricting type parameter `T`
  |
4 | pub struct G1<T: heapuse::HeapSize> {
  |                +++++++++++++++++++
